core/trino-main/src/test/java/io/trino/sql/TestLikeFunctions.java

```diff
@@ -69,8 +80,55 @@ public void testLikeChar()
         assertFalse(likeChar(7L, utf8Slice("foob"), regex));
         assertFalse(likeChar(7L, offsetHeapSlice("foob"), regex));
 
-        assertFunction("cast('foob' as char(6)) LIKE 'f%b__'", BOOLEAN, true);
-        assertFunction("cast('foob' as char(7)) LIKE 'f%b__'", BOOLEAN, false);
+        // pattern shorter than value length
+        assertFunction("CAST('foo' AS char(6)) LIKE 'foo'", BOOLEAN, false);
+        assertFunction("CAST('foo' AS char(6)) LIKE 'foo  '", BOOLEAN, true);
```

Comment:
The pattern is missing one space for this to be `true`.

Reply:
fixed. there is also a case with one more space below